earth2studio/data/jpss.py

```diff
@@ -0,0 +1,652 @@
+
+        Returns
+        -------
+        xr.DataArray
+            Data array containing the requested JPSS data
+        """
+        if self.fs is None:
+            raise ValueError(
+                "File store is not initialized! If you are calling this function directly, make sure the data source is initialized inside the async loop!"
+            )
+
+        # Prepare data inputs
+        time, variable = prep_data_inputs(time, variable)
+
+        # Validate variables
+        JPSS._validate_variables(variable, self._product_type)
+
+        # Create cache directory if it doesn't exist
+        pathlib.Path(self.cache).mkdir(parents=True, exist_ok=True)
+
+        # Add lat/lon to the variable list
+        extended_variables = list(variable) + ["_lat", "_lon"]
+
+        session = await self.fs.set_session()
+
+        # Determine array dimensions
+        y_size, x_size = self.PRODUCT_DIMENSIONS[self._product_type]
+
+        # Create array with extended variables (original + lat + lon)
+        xr_array = xr.DataArray(
+            data=np.zeros(
+                (len(time), len(extended_variables), y_size, x_size), dtype=np.float32
+            ),
+            dims=["time", "variable", "y", "x"],
+            coords={
+                "time": time,
+                "variable": extended_variables,
+                "y": np.arange(y_size),
+                "x": np.arange(x_size),
+            },
+            attrs={
+                "time_data": {},  # Will store {time_index: actual_timestamp}
+                "time_geo": {},  # Will store {time_index: actual_timestamp}
+                "file_data": {},  # Will store {time_index: filename}
+                "file_geo": {},  # Will store {time_index: filename}
+            },
+        )
+
+        # Fetch data
+        async_tasks = [
+            (i, t, j, v) for i, t in enumerate(time) for j, v in enumerate(variable)
+        ]
+        await tqdm.gather(
+            *map(
+                lambda args: self.fetch_data_wrapper(args, xr_array=xr_array),
+                async_tasks,
+            ),
+            desc="Fetching VIIRS data",
+            disable=(not self._verbose),
+        )
+
+        # Fetch geolocation data
+        async_tasks = [(i, t) for i, t in enumerate(time)]
+        await tqdm.gather(
+            *map(
+                lambda args: self.fetch_geolocation_wrapper(args, xr_array=xr_array),
+                async_tasks,
+            ),
+            desc="Fetching VIIRS geolocation data",
+            disable=(not self._verbose),
+        )
+
+        # Validate that data and geolocation timestamps match are within 1 second
+        # NOTE: This should never happen but I will feel better leaving the check here.
+        # Leaving file names in xarray attrs for any future debugging purposes
+        for i in range(len(time)):
+            time_diff = abs(
+                xr_array.attrs["time_data"][i] - xr_array.attrs["time_geo"][i]
```

Comment:
Are these two separate files or the same? If they are the same, maybe commenting this out but keeping it in the code for reference is good since you said it never actually happens?

Reply:
If they are different files, then I agree with keeping this